add support of TINAMI

shuuji3 opened the pull request:
[TINAMI](https://tinami.com) is one of famous Japanese illustration community for fans of anime and comics like Pixiv.

To use TINAMI API, we need to get TINAMI API key. Could you make a request for TINAMI API? It\'s not difficult and I think it takes only a few minutes. Here is workflow needed:

1. Register to TINAMI.
2. Make a request for TINAMI API.
3. Edit `config/default.js` to add a given key to `Client.APIs.tinami`. That\'s all.

You can easily [register to TINAMI](http://www.tinami.com/login?lang=en) via Twitter account, and fill forms in [API Request Form](www.tinami.com/api/key/form).

I translated **API Request form** and **API Guideline** to English for your reference. Visit here: https://gist.github.com/shuuji3/a7185b39c5c73a2ddbeb54682b1f81c8

You can test TINAMI provider by searching for [@tinami_hot](https://twitter.com/tinami_hot).

I added `xml-js`(a XML parser) to `packages.json`, because TINAMI API only supports XML response.

I'm inexperienced with JavaScript, feel free to give a comment if I made something wrong! 

Thank you.

eramdam commented:
Will take a look later in the day

eramdam commented on src/js/util/providers/tinami.js:
```diff
@@ -0,0 +1,38 @@
+    re: /(?:www.tinami.com\/view|tinami.jp\/)/,
+    default: true,
+    callback: url => {
+      let imageId = url.slice(url.lastIndexOf('/') + 1);
```
I'd prefer if you used the [`URL`](https://developer.mozilla.org/en-US/docs/Web/API/URL) API to get this. It would be less error-prone.

eramdam commented on src/js/util/providers/tinami.js:
```diff
@@ -0,0 +1,38 @@
+        .then(json => {
+          // Quit if there is non-public image or no image
+          if (json.rsp._attributes.stat != 'ok' ||
+              json.rsp.content._attributes.type == 'novel') {
```
Please use `!==` and `===` for comparison

shuuji3 commented:
I've fixed my code following your advice like this:
`url.slice(url.lastIndexOf('/') + 1)`
→ `new URL(url).pathname.split('/').pop()` 
Is it right?

In addition to that, fix the API endpoint for image to use HTTPS protocol. This prevents "Mix content warning" in the console.
